plugins/modules/managed_serviceaccount_rbac_addon.py

```diff
@@ -0,0 +1,280 @@
+    random_string = generate_random_string()
+
+    new_manifest_work_raw = Template(MANIFEST_WORK_TEMPLATE).render(
+        cluster_name=managed_cluster_name,
+        managed_service_account_name=managed_service_account_name,
+        owner_api_version=managed_serviceaccount[
+            'managed_serviceaccount']['api_version'],
+        owner_kind=managed_serviceaccount[
+            'managed_serviceaccount']['kind'],
+        owner_name=managed_serviceaccount[
+            'managed_serviceaccount']['name'],
+        owner_uid=managed_serviceaccount[
+            'managed_serviceaccount']['uid'],
+    )
+
+    new_manifest_work = yaml.safe_load(new_manifest_work_raw)
+
+    role_subject = {'kind': 'ServiceAccount',
+                    'name': managed_service_account_name,
+                    'namespace': managed_service_account_namespace}
+    role_names = []
+    filenames = []
+
+    if not os.path.exists(rbac_template):
+        module.fail_json(
+            msg=f"error: RBAC template file or directory {rbac_template} does not exists!")
+        return None
+
+    if os.path.isdir(rbac_template):
+        names = next(os.walk(rbac_template), (None, None, []))[2]
+        for name in names:
+            filenames.append(f"{rbac_template}/{name}")
+        if len(filenames) == 0:
+            module.fail_json(
+                msg=f"error: RBAC template directory {rbac_template} is empty!")
+            return None
+    else:
+        filenames.append(rbac_template)
+
+    try:
+        for filename in filenames:
+            with open(filename, 'r') as file:
+                docs = yaml.safe_load_all(file)
+                for doc in docs:
+                    if doc['kind'] in ['Role', 'ClusterRole']:
+                        role_names.append(doc['metadata']['name'])
+                        doc['metadata']['name'] = f"{doc['metadata']['name']}-{random_string}"
+                    elif doc['kind'] in ['RoleBinding', 'ClusterRoleBinding']:
+                        doc['metadata']['name'] = f"{doc['metadata']['name']}-{random_string}"
+                        if doc['roleRef']['name'] in role_names:
+                            doc['roleRef']['name'] = f"{doc['roleRef']['name']}-{random_string}"
+                        if 'subjects' in doc.keys():
+                            doc['subjects'].append(role_subject)
+                        else:
+                            doc['subjects'] = []
+                            doc['subjects'].append(role_subject)
+
+                    new_manifest_work['spec']['workload']['manifests'].append(
+                        doc)
+    except Exception:
+        module.fail_json(
+            msg=f"error: invalid RBAC template file {filename}")
+
+    manifest_work_api = hub_client.resources.get(
+        api_version='work.open-cluster-management.io/v1',
+        kind='ManifestWork',
+    )
+
+    manifest_work = manifest_work_api.create(new_manifest_work)
```

Comment:
we may want to wait for manifestwork to complete applying

Reply:
The wait is already there:

```
   if wait:
        wait_for_manifestwork_available(
            module, hub_client, manifest_work, timeout)
```